### PR TITLE
Now always include Keycloak in the build + development scripts...

### PR DESCRIPTION
**Pull request by davidfestal**
Now always include Keycloak in the build + development scripts since Keycloak is now enabled / disabled at runtime through a property.

Provide an environment variable to easily enable / disable Keycloak during Minishift deployment. 

Also clean the build and update the readme accordingly. 

Signed-off-by: David Festal <dfestal@redhat.com>

### CODE REVIEW

File: dev-scripts/env-for-minishift [hunk 1]
```diff
@@ -7,3 +7,4 @@ export OPENSHIFT_NAMESPACE_URL=${OPENSHIFT_NAMESPACE_URL:-"${CHE_OPENSHIFT_PROJE
 export CHE_LOG_LEVEL=${CHE_LOG_LEVEL:-"DEBUG"}
 export CHE_DEBUGGING_ENABLED=${CHE_DEBUGGING_ENABLED:-"true"}
 export FABRIC8_ONLINE_PATH=${FABRIC8_ONLINE_PATH:-"${HOME}/github/fabric8-online/"}
+export CHE_KEYCLOAK_DISABLED=${CHE_KEYCLOAK_DISABLED:-"true"}
```
Comment (ibuziuk) [inline dev-scripts/env-for-minishift]: @davidfestal @l0rd should we really disable keycloak on minishift by default ?

Comment (l0rd) [inline dev-scripts/env-for-minishift]: @ibuziuk I requested to @davidfestal to do so. Let me try to explain why: I would like to make it easy for a user to run Che on minishift. And I'm not talking about a Red Hat developer but for anyone that want to try it out. It should be able to use it even if he has not registered on OSIO.

Comment (ibuziuk) [inline dev-scripts/env-for-minishift]: @l0rd makes sense +1

**Comment by davidfestal** [comment]
@l0rd @ibuziuk are you OK to merge ?

**Comment by ibuziuk** [comment]
@davidfestal I'm ok but I still have problems with running it on minishift. Che-server is deployed correctly but when I navigate to route url I keep getting 504 error :/ :

```
  | Using embedded assembly...
-- | --
  | WARN: parent dir is not writeable, CHE_LOCAL_CONF_DIR will be set to /data/conf
  | Using embedded che.properties... Copying template to /data/conf/che.properties
  | Listening for transport dt_socket at address: 8000
  | 2017-07-13 13:05:06,077[main]             [INFO ] [o.a.c.s.VersionLoggerListener 89]    - Server version:        Apache Tomcat/8.5.11
  | 2017-07-13 13:05:06,080[main]             [INFO ] [o.a.c.s.VersionLoggerListener 91]    - Server built:          Jan 10 2017 21:02:52 UTC
  | 2017-07-13 13:05:06,081[main]             [INFO ] [o.a.c.s.VersionLoggerListener 93]    - Server number:         8.5.11.0
  | 2017-07-13 13:05:06,081[main]             [INFO ] [o.a.c.s.VersionLoggerListener 95]    - OS Name:               Linux
  | 2017-07-13 13:05:06,081[main]             [INFO ] [o.a.c.s.VersionLoggerListener 97]    - OS Version:            4.4.27-boot2docker
  | 2017-07-13 13:05:06,082[main]             [INFO ] [o.a.c.s.VersionLoggerListener 99]    - Architecture:          amd64
  | 2017-07-13 13:05:06,082[main]             [INFO ] [o.a.c.s.VersionLoggerListener 101]   - Java Home:             /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.131-3.b12.el7_3.x86_64/jre
  | 2017-07-13 13:05:06,082[main]             [INFO ] [o.a.c.s.VersionLoggerListener 103]   - JVM Version:           1.8.0_131-b12
  | 2017-07-13 13:05:06,083[main]             [INFO ] [o.a.c.s.VersionLoggerListener 105]   - JVM Vendor:            Oracle Corporation
  | 2017-07-13 13:05:06,083[main]             [INFO ] [o.a.c.s.VersionLoggerListener 107]   - CATALINA_BASE:         /home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat
  | 2017-07-13 13:05:06,083[main]             [INFO ] [o.a.c.s.VersionLoggerListener 109]   - CATALINA_HOME:         /home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat
  | 2017-07-13 13:05:06,084[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dnop
  | 2017-07-13 13:05:06,084[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
  | 2017-07-13 13:05:06,084[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+UseG1GC
  | 2017-07-13 13:05:06,084[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+UseStringDeduplication
  | 2017-07-13 13:05:06,085[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:MinHeapFreeRatio=20
  | 2017-07-13 13:05:06,085[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:MaxHeapFreeRatio=40
  | 2017-07-13 13:05:06,085[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:MaxRAM=700m
  | 2017-07-13 13:05:06,085[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+UnlockExperimentalVMOptions
  | 2017-07-13 13:05:06,086[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+UseCGroupMemoryLimitForHeap
  | 2017-07-13 13:05:06,086[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dche.docker.network=bridge
  | 2017-07-13 13:05:06,086[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dport.http=8080
  | 2017-07-13 13:05:06,086[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dche.home=/home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT
  | 2017-07-13 13:05:06,086[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dche.logs.dir=/logs/
  | 2017-07-13 13:05:06,087[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dche.logs.level=INFO
  | 2017-07-13 13:05:06,087[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djuli-logback.configurationFile=file:/home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat/conf/tomcat-logger.xml
  | 2017-07-13 13:05:06,087[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
  | 2017-07-13 13:05:06,087[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources
  | 2017-07-13 13:05:06,087[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -agentlib:jdwp=transport=dt_socket,address=8000,server=y,suspend=n
  | 2017-07-13 13:05:06,088[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcom.sun.management.jmxremote
  | 2017-07-13 13:05:06,088[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcom.sun.management.jmxremote.ssl=false
  | 2017-07-13 13:05:06,088[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcom.sun.management.jmxremote.authenticate=false
  | 2017-07-13 13:05:06,088[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dche.local.conf.dir=/data/conf
  | 2017-07-13 13:05:06,089[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcatalina.base=/home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat
  | 2017-07-13 13:05:06,089[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcatalina.home=/home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat
  | 2017-07-13 13:05:06,089[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djava.io.tmpdir=/home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat/temp
  | 2017-07-13 13:05:06,214[main]             [INFO ] [o.a.c.http11.Http11NioProtocol 525]  - Initializing ProtocolHandler ["http-nio-8080"]
  | 2017-07-13 13:05:06,235[main]             [INFO ] [o.a.t.util.net.NioSelectorPool 67]   - Using a shared selector for servlet write/read
  | 2017-07-13 13:05:06,241[main]             [INFO ] [o.a.catalina.startup.Catalina 617]   - Initialization processed in 525 ms
  | 2017-07-13 13:05:06,272[main]             [INFO ] [c.m.JmxRemoteLifecycleListener 336]  - The JMX Remote Listener has configured the registry on port 32001 and the server on port 32101 for the Platform server
  | 2017-07-13 13:05:06,273[main]             [INFO ] [o.a.c.core.StandardService 416]      - Starting service Catalina
  | 2017-07-13 13:05:06,274[main]             [INFO ] [o.a.c.core.StandardEngine 259]       - Starting Servlet Engine: Apache Tomcat/8.5.11
  | 2017-07-13 13:05:12,228[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 923]        - Deploying web application archive /home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat/webapps/dashboard.war
  | 2017-07-13 13:05:17,554[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 987]        - Deployment of web application archive /home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat/webapps/dashboard.war has finished in 5,325 ms
  | 2017-07-13 13:05:17,556[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 923]        - Deploying web application archive /home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat/webapps/wsmaster.war
  | 2017-07-13 13:05:29,476[ost-startStop-1]  [INFO ] [o.f.c.i.d.DbSupportFactory 44]       - Database: jdbc:h2:che (H2 1.4)
  | 2017-07-13 13:05:29,564[ost-startStop-1]  [INFO ] [o.f.c.i.util.VersionPrinter 44]      - Flyway 4.0.3 by Boxfuse
  | 2017-07-13 13:05:29,565[ost-startStop-1]  [INFO ] [o.f.c.i.d.DbSupportFactory 44]       - Database: jdbc:h2:che (H2 1.4)
  | 2017-07-13 13:05:29,611[ost-startStop-1]  [INFO ] [i.f.CustomSqlMigrationResolver 153]  - Searching for sql scripts in locations [classpath:che-schema]
  | 2017-07-13 13:05:29,663[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbValidate 44]       - Successfully validated 11 migrations (execution time 00:00.053s)
  | 2017-07-13 13:05:29,679[ost-startStop-1]  [INFO ] [o.f.c.i.m.MetaDataTableImpl 44]      - Creating Metadata table: "PUBLIC"."schema_version"
  | 2017-07-13 13:05:29,695[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Current version of schema "PUBLIC": << Empty Schema >>
  | 2017-07-13 13:05:29,696[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.0.0.8.1 - 1__init.sql
  | 2017-07-13 13:05:29,848[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.0.0.9.1 - 1__add_index_on_workspace_temporary.sql
  | 2017-07-13 13:05:29,862[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.0.0.9.2 - 2__update_local_links_in_environments.sql
  | 2017-07-13 13:05:29,875[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.2.0.1 - 1__increase_project_attributes_values_length.sql
  | 2017-07-13 13:05:29,903[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.4.0.1 - 1__drop_user_to_account_relation.sql
  | 2017-07-13 13:05:29,962[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.4.0.2 - 2__create_missed_account_indexes.sql
  | 2017-07-13 13:05:29,975[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.6.0.1 - 1__add_exec_agent_where_terminal_agent_is_present.sql
  | 2017-07-13 13:05:29,997[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.7.0.1 - 1__add_factory.sql
  | 2017-07-13 13:05:30,087[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.7.0.2 - 2__remove_match_policy.sql
  | 2017-07-13 13:05:30,148[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.8.0.1 - 1__add_foreigh_key_indexes.sql
  | 2017-07-13 13:05:30,190[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "PUBLIC" to version 5.11.0.1 - 1__optimize_user_search.sql
  | 2017-07-13 13:05:30,216[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Successfully applied 11 migrations to schema "PUBLIC" (execution time 00:00.538s).
  | 2017-07-13 13:05:34,619[ost-startStop-1]  [WARN ] [o.e.c.a.w.s.stack.StackLoader 87]    - Reload stacks on start is deprecated policy, and it will be removed soon
  | 2017-07-13 13:05:35,014[ost-startStop-1]  [ERROR] [o.e.c.a.w.s.stack.StackLoader 101]   - Failed to store stacks from 'che-in-che.json'
  | 2017-07-13 13:05:35,017[ost-startStop-1]  [INFO ] [o.e.c.a.w.s.stack.StackLoader 104]   - Stacks initialization finished
  | 2017-07-13 13:05:35,085[ost-startStop-1]  [INFO ] [o.e.c.a.m.s.r.RecipeLoader 75]       - Recipes initialization finished
  | 2017-07-13 13:05:35,128[ted-scheduler-1]  [INFO ] [ockerAbandonedResourcesCleaner 124]  - List containers registered in the api: []
  | 2017-07-13 13:05:35,161[ost-startStop-1]  [INFO ] [s.KeycloakAuthenticationFilter 46]   - Keycloak is disabled
  | 2017-07-13 13:05:36,288[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 987]        - Deployment of web application archive /home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat/webapps/wsmaster.war has finished in 18,732 ms
  | 2017-07-13 13:05:36,308[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 923]        - Deploying web application archive /home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat/webapps/ROOT.war
  | 2017-07-13 13:05:47,887[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 987]        - Deployment of web application archive /home/user/eclipse-che-5.15.0-fabric8-SNAPSHOT/tomcat/webapps/ROOT.war has finished in 11,578 ms
  | 2017-07-13 13:05:47,899[main]             [INFO ] [o.a.c.http11.Http11NioProtocol 570]  - Starting ProtocolHandler [http-nio-8080]
  | 2017-07-13 13:05:47,922[main]             [INFO ] [o.a.catalina.startup.Catalina 668]   - Server startup in 41679 ms


```

**Comment by ibuziuk** [comment]
@davidfestal let's push it but I still have problems with keycloak enabled:


```
2017-07-13 13:19:45,296[nio-8080-exec-4]  [ERROR] [c.r.che.valve.UserAuthValve 144]     - Exception during Keycloak settings retrieval
--
  | java.net.ConnectException: Connection timed out (Connection timed out)
  | at java.net.PlainSocketImpl.socketConnect(Native Method)
  | at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
  | at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
  | at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
  | at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
  | at java.net.Socket.connect(Socket.java:589)
  | at java.net.Socket.connect(Socket.java:538)
  | at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
  | at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
  | at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
  | at sun.net.www.http.HttpClient.<init>(HttpClient.java:242)
  | at sun.net.www.http.HttpClient.New(HttpClient.java:339)
  | at sun.net.www.http.HttpClient.New(HttpClient.java:357)
  | at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1202)
  | at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1138)
  | at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1032)
  | at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:966)
  | at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1546)
  | at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)
  | at com.redhat.che.valve.UserAuthValve.retrieveKeycloakDisabledSetting(UserAuthValve.java:128)
  | at com.redhat.che.valve.UserAuthValve.isKeycloakDisabled(UserAuthValve.java:116)
  | at com.redhat.che.valve.UserAuthValve.authenticate(UserAuthValve.java:56)
  | at org.keycloak.adapters.tomcat.KeycloakAuthenticatorValve.doAuthenticate(KeycloakAuthenticatorValve.java:54)
  | at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:556)
  | at org.keycloak.adapters.tomcat.AbstractKeycloakAuthenticatorValve.invoke(AbstractKeycloakAuthenticatorValve.java:187)
  | at org.apache.catalina.valves.rewrite.RewriteValve.invoke(RewriteValve.java:580)
  | at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
  | at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
  | at ch.qos.logback.access.tomcat.LogbackValve.invoke(LogbackValve.java:256)
  | at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:677)
  | at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
  | at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:349)
  | at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:783)
  | at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
  | at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:798)
  | at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1434)
  | at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
  | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
  | at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
  | at java.lang.Thread.run(Thread.java:748)
  | 2017-07-13 13:20:15,202[io-8080-exec-10]  [ERROR] [c.r.che.valve.UserAuthValve 144]     - Exception during Keycloak settings retrieval
  | java.net.ConnectException: Connection timed out (Connection timed out)
  | at java.net.PlainSocketImpl.socketConnect(Native Method)
  | at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
  | at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
  | at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
  | at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
  | at java.net.Socket.connect(Socket.java:589)
  | at java.net.Socket.connect(Socket.java:538)
  | at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
  | at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
  | at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
  | at sun.net.www.http.HttpClient.<init>(HttpClient.java:242)
  | at sun.net.www.http.HttpClient.New(HttpClient.java:339)
  | at sun.net.www.http.HttpClient.New(HttpClient.java:357)
  | at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1202)
  | at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1138)
  | at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1032)
  | at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:966)
  | at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1546)
  | at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)
  | at com.redhat.che.valve.UserAuthValve.retrieveKeycloakDisabledSetting(UserAuthValve.java:128)
  | at com.redhat.che.valve.UserAuthValve.isKeycloakDisabled(UserAuthValve.java:116)
  | at com.redhat.che.valve.UserAuthValve.authenticate(UserAuthValve.java:56)
  | at org.keycloak.adapters.tomcat.KeycloakAuthenticatorValve.doAuthenticate(KeycloakAuthenticatorValve.java:54)
  | at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:556)
  | at org.keycloak.adapters.tomcat.AbstractKeycloakAuthenticatorValve.invoke(AbstractKeycloakAuthenticatorValve.java:187)
  | at org.apache.catalina.valves.rewrite.RewriteValve.invoke(RewriteValve.java:580)
  | at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
  | at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
  | at ch.qos.logback.access.tomcat.LogbackValve.invoke(LogbackValve.java:256)
  | at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:677)
  | at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
  | at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:349)
  | at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:783)
  | at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
  | at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:798)
  | at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1434)
  | at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
  | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
  | at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
  | at java.lang.Thread.run(Thread.java:748)

```

**Comment by ibuziuk** [comment]
@davidfestal status api works for me `/api/system/state` and readiness probe passes, but I can not open dashboard for some reason - now looking into filters code...

File: README.md [hunk 2]
```diff
@@ -182,6 +178,14 @@ You can even override the name of the docker image with the `CHE_IMAGE_REPO` var
 bash -c "export CHE_IMAGE_REPO=rhche/che-server ; export CHE_IMAGE_TAG=nightly ; dev-scripts/minishift_deploy.sh"
 ```
 
+By default, the docker image of the deployed to Minishift with the Keycloak integration *disabled*.
```
Comment (l0rd) [inline README.md]: s/the docker image of the deployed to Minishfit with/Che is deployed on minishift with/

**Comment by davidfestal** [comment]
@mario Readme fixes done

**Pull request merged**